lib/SmartyPlugins/LMSSmartyPlugins.php

```diff
@@ -89,13 +89,16 @@ public static function currencySelectionFunction(array $params, $template)
         $elementname = isset($params['elementname']) ? $params['elementname'] : 'currency';
         $selected = isset($params['selected']) && isset($GLOBALS['CURRENCIES'][$params['selected']])
             ? $params['selected'] : null;
+        $defaultSelected = Localisation::getCurrentCurrency();
         $locked = isset($params['locked']) && $params['locked'];
         if (function_exists('get_currency_value') && !$locked) {
-            $result = '<select name="' . $elementname . '" ' . self::tipFunction(array('text' => 'Select currency'), $template)
+            $result = '<select class="'. (!$selected ? 'lms-ui-warning' : '')
+                .'" name="' . $elementname . '" '
+                . self::tipFunction(array('text' => !$selected ? 'Select currency and save' : 'Select currency'), $template)
                 . (isset($params['form']) ? ' form="' . $params['form'] . '"' : '') . '>';
             foreach ($GLOBALS['CURRENCIES'] as $currency) {
                 $result .= '<option value="' . $currency . '"'
-                    . ($currency == $selected ? ' selected' : '') . '>' . $currency . '</option>';
+                    . ($currency == $selected || $currency == $defaultSelected ? ' selected' : '') . '>' . $currency . '</option>';
             }
```

Comment:
Czy to nie zrobi problemu, gdy wybraną walutą będzie np. EUR, a domyślną systemową PLN?
Wtedy nagle waluta taryfy przeskoczy z wartości dotychczasowej EUR na PLN?
Czy się mylę? :)
Moim zdaniem warunek `$currency == $defaultSelected` powinien jeszcze być powiązany z drugim warunkiem, że $currency  jest puste?

Reply:
> Czy to nie zrobi problemu, gdy wybraną walutą będzie np. EUR, a domyślną systemową PLN?
> Wtedy nagle waluta taryfy przeskoczy z wartości dotychczasowej EUR na PLN?

Zgadza się. Nie przetestowałem tego wariantu. Puściłem poprawkę, która ten problem eliminuje.